### PR TITLE
Feedbot Refactor

### PR DESCRIPTION
**Pull request by jarrel-b**
Noticed it was a bit hard to add new events to the bot and also hard to create more personalized messages, so I made some changes that should make it easier to extend the bot.

#### Few new concepts
A `Query` is a mapping of data surrounding an event that is used to match to a `Post`. A `Post` contains a `rule` which is a set of criteria that is used to evaluate a query and also a `poster` which is a way to handle a matched query. 
When an event is received, the bot builds a query and then finds the first matching `Post` for that query and calls its `poster`.

This should make it easy to render very customizable messages i.e.:
```
{"CryptoCovenHolder": true, "isNewUserEvent": true, "username": "Jarrel"} => "A new witch, Jarrel, joined Gallery!"
{"BeatFoundryHolder": true, "isNewUserEvent": true, "username": "Jarrel"} => "A new beatmaker, Jarrel, joined Gallery!"
{"isNewUserEvent": true, "username": "Jarrel"} => "Jarrel joined Gallery!"
```
The bot only has the `DiscordPoster` which handles the templating and sending of a post.

#### `sent` column
I added a `sent` column to each of the event tables to mark if an event was used for a feed. Thinking this will be useful for diffs i.e. "Jarrel followed 12 more people", "Jarrel added 5 new NFTs to their collection", etc.

### CODE REVIEW

**Comment by Robinnnnn** [comment]
this PR is a work of art 🎨 🎨 🎨

**Comment by kaitoo1** [comment]
customizable messages and diffs are super cool. can see them adding a lot of texture to the feed, great idea 🔥

File: service/event/cloudtask/cloudtask.go [hunk 1]
```diff
@@ -46,10 +46,10 @@ func createTaskForService(ctx context.Context, queuePath string, scheduleOn time
 			ScheduleTime: timestamppb.New(scheduleOn),
 			MessageType: &taskspb.Task_AppEngineHttpRequest{
 				AppEngineHttpRequest: &taskspb.AppEngineHttpRequest{
-					HttpMethod:       taskspb.HttpMethod_POST,
-					AppEngineRouting: &taskspb.AppEngineRouting{Service: service},
-					RelativeUri:      uri,
-					Headers:          headers,
+					HttpMethod: taskspb.HttpMethod_POST,
+					// XXX: AppEngineRouting: &taskspb.AppEngineRouting{Service: service},
```
Comment (kaitoo1) [inline service/event/cloudtask/cloudtask.go:50]: do we need this?

**Review by kaitoo1 — APPROVED** [review]

File: service/event/cloudtask/cloudtask.go (hunk 1 above)
Comment (jarrel-b) [reply service/event/cloudtask/cloudtask.go:50]: Oh yea need to uncomment this out, you need to comment out this line atm to test the bot locally

**Review by bennycio — APPROVED** [review]
Super clean!

**Pull request merged**